instrumentation-genai/opentelemetry-instrumentation-openai-v2/tests/test_async_embeddings.py

```diff
@@ -0,0 +1,311 @@
+    span: ReadableSpan,
+    request_model: str,
+    response,
+):
+    """Assert that the span contains all required attributes for embeddings operation"""
+    # Use the common assertion function
+    assert_all_attributes(
+        span,
+        request_model,
+        response_id=None,  # Embeddings don't have a response ID
+        response_model=response.model,
+        input_tokens=response.usage.prompt_tokens,
+        output_tokens=None,  # Embeddings don't have separate output tokens
+        operation_name="embeddings",
+        server_address="api.openai.com",
+    )
+
+    # Assert embeddings-specific attributes
+    if (
+        hasattr(span, "attributes")
+        and "gen_ai.embeddings.dimension.count" in span.attributes
+    ):
+        # If dimensions were specified, verify that they match the actual dimensions
+        assert span.attributes["gen_ai.embeddings.dimension.count"] == len(
+            response.data[0].embedding
+        )
+
+    # Assert tokens are correctly recorded
+    assert (
+        span.attributes[GenAIAttributes.GEN_AI_USAGE_INPUT_TOKENS]
+        == response.usage.prompt_tokens
+    )
+
+
+def assert_all_attributes(
+    span: ReadableSpan,
+    request_model: str,
+    response_id: str = None,
+    response_model: str = None,
+    input_tokens: Optional[int] = None,
+    output_tokens: Optional[int] = None,
+    operation_name: str = "embeddings",
+    server_address: str = "api.openai.com",
+):
+    """Assert common attributes on the span"""
+    assert span.name == f"{operation_name} {request_model}"
+    assert (
+        operation_name
+        == span.attributes[GenAIAttributes.GEN_AI_OPERATION_NAME]
+    )
+    assert (
+        GenAIAttributes.GenAiSystemValues.OPENAI.value
+        == span.attributes[GenAIAttributes.GEN_AI_SYSTEM]
+    )
+    assert (
+        request_model == span.attributes[GenAIAttributes.GEN_AI_REQUEST_MODEL]
+    )
+
+    if response_model:
+        assert (
+            response_model
+            == span.attributes[GenAIAttributes.GEN_AI_RESPONSE_MODEL]
+        )
+    else:
+        assert GenAIAttributes.GEN_AI_RESPONSE_MODEL not in span.attributes
+
+    if response_id:
+        assert (
+            response_id == span.attributes[GenAIAttributes.GEN_AI_RESPONSE_ID]
+        )
+    else:
+        assert GenAIAttributes.GEN_AI_RESPONSE_ID not in span.attributes
+
+    if input_tokens:
+        assert (
+            input_tokens
+            == span.attributes[GenAIAttributes.GEN_AI_USAGE_INPUT_TOKENS]
+        )
+    else:
+        assert GenAIAttributes.GEN_AI_USAGE_INPUT_TOKENS not in span.attributes
+
+    if output_tokens:
+        assert (
+            output_tokens
+            == span.attributes[GenAIAttributes.GEN_AI_USAGE_OUTPUT_TOKENS]
+        )
+    else:
+        assert (
+            GenAIAttributes.GEN_AI_USAGE_OUTPUT_TOKENS not in span.attributes
+        )
+
+    assert server_address == span.attributes[ServerAttributes.SERVER_ADDRESS]
```

Comment:
I think we could also assert server port and gen_ai.request.encoding_formats here